### DISCUSSION TITLE
Datafile conversion

### DISCUSSION THREAD

**Discussion by NoSuchCommand** [sysstat/sysstat#361] [Q&A]
Hello,

I have been looking for a way to extract a specific time range from a `sar` datafile (written with `sar -o`) and keep the result as a datafile (queryable through `sar -f`). I thought it was possible to do so with `sadf`, using `-s` and `-e`, but it appears that even the raw format (`sadf -r`) is a text format.

Would there be any way to manage this ? Even a _hacky_ one ?

**Comment by sysstat** [comment]
Hi,
No, you cannot extract a specific time range from a sar datafile and save it as a binary file. Data are always extracted and displayed in a text format.

**Comment by NoSuchCommand** [comment]
Thank you very much for your answer :-)

Sad news though. Do you think such a feature could be a good candidate for an RFE?

I understand that `sadf` stands for _display format_, but my intent is not to display. It's more about filtering a datafile to just keep a specific time range, preserving the raw datafile format. This could be useful, for example, when sharing a datafile with a team that does not care about anything else than the time span where something interesting happened, or to store the datafile as a reference alongside a documentation.

Of course, one can think about sharing the data as already exported in any of the formats supported by `sadf` but, doing so, the recipient loses the ability to manipulate them again at her convenience with `sar` or `sadf`, and has to develop her own tools to leverage them.

I guess that, since `sar` can read and write a datafile with `-f` and `-o` respectively (which, I know, are currently mutually exclusive), it could be possible to implement a way of doing something along the lines of `sar -A -f input.data -o output.data -s 11:00 -e 12:00`?

**Reply by natoscott** [reply]
sadf can export to the Performance Co-Pilot format (https://pcp.io) which does support this concept via the pmlogextract(1) tool - perhaps that's an option here?

There are some plans to extend sar to be able to read PCP archives in time as well, but no working code for that just yet.

**Reply by NoSuchCommand** [reply]
I was just exploring this path, but I was struggling to find the command among all the PCP tools. Thanks for the shortcut ;-)
